pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
> Do we support selector inheritance as I described? Inherit the labels by default?

I understood this to mean that we apply use the labels in the selector, is there more to it than that? (we still apply the labels to all objects generated)